tgui/packages/tgui/interfaces/LuaEditor/index.js

```diff
@@ -60,21 +63,20 @@ export class LuaEditor extends Component {
       'modal',
       noStateYet ? 'states' : null
     );
-    const [activeTab, setActiveTab] = useLocalState(
-      this.context,
-      'activeTab',
-      showGlobalTable ? 'globals' : 'tasks'
-    );
-    const [input, setInput] = useLocalState(this.context, 'scriptInput', '');
-    const [shouldUpdateScroll, setShouldUpdateScroll] = useLocalState(
-      this.context,
-      'shouldUpdateScroll',
-      false
-    );
+    const [, setViewedChunk] = useLocalState(this.context, 'viewedChunk');
+    let { shouldUpdateScroll, forceModal, forceViewChunk } = data;
     if (shouldUpdateScroll) {
-      setShouldUpdateScroll(false);
+      shouldUpdateScroll = 0;
       setTimeout(this.handleSectionScroll, 0);
     }
```

Comment:
I think the entire way of dealing with scroll is way too overcomplicated, since you can write the entire effect in `componentDidMount()`

Reply:
I didn't quite get the behavior I was looking for putting the code in `componentDidMount`, but I did in `componentDidUpdate`